fieldservice_account/models/account_invoice.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    fsm_order_id = fields.Many2one('fsm.order', string='FSM Order')
+
+    @api.model
+    def create(self, vals):
+        order = self.env['fsm.order'].browse(vals.get('fsm_order_id'))
+        if order:
+            if order.location_id.analytic_account_id:
+                vals['account_analytic_id'] = order.location_id.\
+                    analytic_account_id.id
+            else:
+                raise ValidationError("No analytic account" +
+                                      " set on the order's Location")
+
+        return super().create(vals)
```

Comment:
`return super(AccountInvoiceLine, self).create(vals)`

Reply:
Fixed in new commit